File: tests/sample_code.js
// Sample JavaScript file for testing TODO detection

function main() {
    console.log("Hello, World!");
    
    // TODO: Add input validation
    const userInput = prompt("Enter your name:");
    
    // TODO: Handle null/undefined input
    const processed = processInput(userInput);
    
    // FIXME: This function doesn't exist yet
    displayResult(processed);
}

function processInput(input) {
    // TODO: Implement proper string sanitization
    return input ? input.trim().toLowerCase() : '';
}

// TODO: Implement the displayResult function
// TODO: Add error handling throughout the application

main();

File: tests/sample_code.py
#!/usr/bin/env python3
"""
Sample Python file for testing TODO detection.
"""

def main():
    """Main function."""
    print("Hello, World!")
    
    # TODO: Add proper error handling
    user_input = input("Enter your name: ")
    
    # TODO: Validate input before processing
    process_input(user_input)
    
    # FIXME: This should use a better algorithm
    result = calculate_something()
    
    return result


def process_input(user_input):
    """Process user input."""
    # TODO: Implement input sanitization
    processed = user_input.strip().lower()
    return processed


def calculate_something():
    """Calculate something important."""
    # TODO: Replace with actual calculation
    return 42


# TODO: Add comprehensive unit tests for all functions
if __name__ == "__main__":
    main()
